src/NzbDrone.Core/DecisionEngine/Specifications/UpgradableSpecification.cs

```diff
@@ -142,17 +136,20 @@ public bool IsRevisionUpgrade(QualityModel currentQuality, QualityModel newQuali
             return false;
         }
 
-        public bool IsUpgradeAllowed(QualityProfile qualityProfile, QualityModel currentQuality, QualityModel newQuality)
+        public bool IsUpgradeAllowed(QualityProfile qualityProfile, QualityModel currentQuality, List<CustomFormat> currentCustomFormats, QualityModel newQuality, List<CustomFormat> newCustomFormats)
         {
             var isQualityUpgrade = new QualityModelComparer(qualityProfile).Compare(newQuality, currentQuality) > 0;
+            var isCustomFormatUpgrade = qualityProfile.CalculateCustomFormatScore(newCustomFormats) > qualityProfile.CalculateCustomFormatScore(currentCustomFormats);
+            var t1 = qualityProfile.CalculateCustomFormatScore(newCustomFormats);
+            var t2 = qualityProfile.CalculateCustomFormatScore(currentCustomFormats);
 
-            if (isQualityUpgrade && qualityProfile.UpgradeAllowed)
+            if ((isQualityUpgrade || isCustomFormatUpgrade) && qualityProfile.UpgradeAllowed)
             {
-                _logger.Debug("At least one profile allows upgrading");
+                _logger.Debug("Quality profile allows upgrading");
                 return true;
             }
 
-            if (isQualityUpgrade && !qualityProfile.UpgradeAllowed)
+            if ((isQualityUpgrade || isCustomFormatUpgrade) && !qualityProfile.UpgradeAllowed)
```

Comment:
Left this as a separate check so lower/same quality/CF score aren't rejected incorrectly as they'll be handled with another spec.